Fix/node account id freeze error

Blockchain-Oracle opened the pull request.

Blockchain-Oracle committed:
fix: automatically set node account IDs from wallet network

Fixes the "nodeAccountId must be set" error that users were experiencing
when calling submitMessageToTopic() or executing manual transactions.

Users encountered two related errors:
1. "nodeAccountId must be set or client must be provided with freezeWith"
2. "Attempting to execute a transaction against nodes, which are not
   included in the Client's node list"

Modified executeTransaction() to automatically set node account IDs from
the signer's network configuration before freezing transactions. This
ensures transactions always have valid nodes from the wallet's network.

- src/index.ts: Added automatic node account ID assignment in executeTransaction()
- scripts/generate-cjs-stub.mjs: Created missing CJS build script

The fix retrieves nodes from signer.getNetwork() and uses the first 3
nodes as the transaction's node account IDs, ensuring compatibility
with the wallet's network configuration.

Blockchain-Oracle committed:
fix: enhance transaction validation and error handling

Improves the transaction execution process by adding checks for the signer and ensuring that node account IDs are retrieved correctly from the signer's network. This prevents potential errors related to missing signer or node account IDs during transaction freezing.

- Added error handling for missing signer and network availability.
- Updated logic to use node account IDs from the signer's network, ensuring transactions have valid nodes.

This change aims to enhance user experience by providing clearer error messages and ensuring proper transaction setup.

kantorcodes commented on src/index.ts:
```diff
@@ -214,6 +214,32 @@ class HashinalsWalletConnectSDK {
+
+    // Check if nodeAccountIds is null or empty array
+    const nodeAccountIds = tx.nodeAccountIds || [];
+    if (nodeAccountIds.length === 0) {
```
This looks good, just confirming have you tested this locally?

Blockchain-Oracle replied:
YES tested locally with yalc ,  refer to this codebase https://github.com/Blockchain-Oracle/test-hashinal-wc

kantorcodes commented:
Could you ensure commits are GPG signed?

kantorcodes commented on src/index.ts:
```diff
@@ -214,6 +214,32 @@ class HashinalsWalletConnectSDK {
     );
+
+    // Ensure the transaction has node account IDs set before freezing
+    // This prevents the "nodeAccountId must be set" error
+    if (!signer) {
+      throw new Error('No signer available. Please ensure wallet is connected.');
+    }
+
```
@Blockchain-Oracle

Blockchain-Oracle committed:
chore: update hedera-wallet-connect to version 2.0.4-canary.3ca04e9.0

Bumps the @hashgraph/hedera-wallet-connect dependency to the latest canary version, ensuring compatibility with recent changes and improvements. Updates the pnpm lock file accordingly to reflect this change.

Blockchain-Oracle commented:
> Could you ensure commits are GPG signed?

Yes the commit are verified hence GPG signed confirmed ✅

Blockchain-Oracle commented:
Thank you @kantorcodes
